### ISSUE TITLE
Rename doesn't de-var-ify vars

### ISSUE THREAD

**Issue by SLaks** [dotnet/roslyn#5872]
``` C#
class X {
    X() {
        var a = 2;
    }
}
```

Rename `X` to `var`.

It reports an unresolvable conflict, instead of changing `var` in the declaration to `int`.

**Comment by SLaks** [comment]
Beautifully, if you change `2` to `new X()`, it doesn't report a conflict at all, and instead changes the meaning of the existing `var` token.

**Comment by CyrusNajmabadi** [comment]
Why would you rename something to 'var'?  Ideally we'd just disallow any user types from being named 'var' in the first place.  It'll just cause *waaay* too many problems for people reading the code.

**Comment by SLaks** [comment]
True; this is a rather corner-case issue.

And, you can't do that due to (insane) back-compat requirements.

**Comment by Pilchie** [comment]
You could imagine an expand/reduce pair that converted `var` to an explicit type if possible, but I don't think it's something we would invest in.
